admin/services/pupil-register-v2.service.js

```diff
@@ -11,6 +13,24 @@ const pupilRegisterV2Service = {
    * @return {Promise<*>}
    */
   getPupilRegister: async function (schoolId) {
+    if (!schoolId) {
+      throw new Error('School id not found in session')
+    }
+    const pupilRegisterRedisKey = `pupilRegisterViewData:${schoolId}`
```

Comment:
I am slightly concerned that there isn't a buildKey() method or something.  If you GET the key here; how is it set? Duplicate code?

Reply:
seems like a good opportunity to implement the `redisKeyService` we talked about.

`getPupilRegisterViewDataKey(schoolid: number): string`

Reply:
added